Library/Homebrew/cli/named_args.rb

```diff
@@ -28,25 +34,30 @@ def initialize(
         cask_options: false,
         without_api: false
       )
-        @args = args
+        super(args)
+
         @override_spec = override_spec
         @force_bottle = force_bottle
         @flags = flags
         @cask_options = cask_options
         @without_api = without_api
         @parent = parent
-
-        super(@args)
       end
 
-      attr_reader :parent
-
+      sig { returns(T::Array[Cask::Cask]) }
       def to_casks
-        @to_casks ||= to_formulae_and_casks(only: :cask).freeze
+        @to_casks ||= T.let(
+          to_formulae_and_casks(only: :cask).freeze, T.nilable(T::Array[T.any(Formula, Keg, Cask::Cask)])
+        )
+        T.cast(@to_casks, T::Array[Cask::Cask])
```

Comment:
There are unfortunately a lot of casts in this file. Some of this is due to sorbet limitations (e.g. limited ability to use `partition` as a type guard). I might make a second pass to reduce this, but didn't want to undertake a large refactor in the same PR.